experimental/sev_guest/src/ghcb.rs

```diff
@@ -219,6 +219,20 @@ impl<'a> GhcbProtocol<'a> {
         self.do_vmg_exit()
     }
 
+    /// Read an 8 bit number from an IO port via the IOIO protocol.
+    ///
+    /// See section 4.1.2 in <https://developer.amd.com/wp-content/resources/56421.pdf>.
+    pub fn io_read_u8(&mut self, port: u16) -> Result<u8, &'static str> {
```

Comment:
There are read and write implementations for 32 bit and 8 bit values (8 bit write is further up in the file, 32 bit read is further down). The 8 bit read and write is used by the serial port implementation. The 32 bit implementations will be used for the SEV IO device. I added this 8 bit read because I needed it for the serial port but did not previously think it would be needed.

Reply:
What is SEV IO device doing?
It's not for communicating with the hypervisor, right? (only the serial port is used for this?)

Reply:
It is used as a high-bandwidth comms channel between the untrusted launcher and the trusted runtime in cases where the hypervisor does not have virtio vsock support.

The guest driver is currently implemented using normal IO port writes for configuration and notifying the hypervisor (https://github.com/project-oak/oak/blob/main/experimental/oak_baremetal_simple_io/src/lib.rs#L52-L53). This works without encryption and with SEV, but will not work with SEV-ES or SEV-SNP where these IO reads and writes will cause exceptions/interrupts in the guest. This code will be updated to use the 32bit versions of the GHCB IOIO protocol when running with SEV-ES or SEV-SNP.